app/components/certificate/certificate-display.tsx

```diff
@@ -0,0 +1,118 @@
+import {
+  Flex,
+  Box,
+  HStack,
+  Text,
+  Heading,
+  Tooltip,
+  IconButton,
+  Accordion,
+  AccordionItem,
+  AccordionButton,
+  AccordionIcon,
+  AccordionPanel,
+  useToast,
+} from '@chakra-ui/react';
+import { DownloadIcon, CopyIcon } from '@chakra-ui/icons';
+import type { Certificate } from '@prisma/client';
+
+interface CertificateDisplayProps {
+  certificate: Certificate;
+  isPrivate: Boolean;
+}
+
+export default function CertificateDisplay(props: CertificateDisplayProps) {
+  const { isPrivate, certificate } = props;
+  const toast = useToast();
+  let description: string = '',
+    currentState: string = '',
+    key: string = '';
+
+  switch (isPrivate) {
+    case true:
+      currentState = 'Private';
+      key = certificate.privateKey;
+      description =
+        "Private: Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s";
+      break;
+    case false:
+      currentState = 'Public';
+      key = certificate.certificate;
+      description =
+        "Public: Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s";
+      break;
+  }
+
+  function onCopy() {
+    navigator.clipboard.writeText(key);
+    toast({
+      title: `${currentState} was copied to clipboard`,
+      position: 'bottom-right',
+      status: 'success',
+    });
+  }
+
+  return (
+    <Flex flexDirection="column" gap="5">
+      <Flex flexDirection="column" gap="4">
+        <HStack gap="4">
+          <Heading as="h4" size="sm">
+            {`${currentState} key`}
+          </Heading>
+          <Tooltip label={`Copy ${currentState} Key`}>
```

Comment:
Throughout this file, you append ` Key` to strings.  As I mention above, that doesn't work for a certificate.  I would use the `name`/`title` prop and have it include everything you need: `Copy Certificate` and `Copy Private Key` where `title='Certificate'` and `title='Private Key'`.